public/app/features/transformers/renameByMapping/renameByMapping.ts

```diff
@@ -0,0 +1,106 @@
+import { map } from 'rxjs/operators';
+
+import {
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  Field,
+  FieldType,
+  SynchronousDataTransformerInfo,
+} from '@grafana/data';
+import { getFieldDisplayName } from '@grafana/data/src/field/fieldState';
+import { getTemplateSrv, TemplateSrv } from 'app/features/templating/template_srv';
+
+export interface RenameByMappingTransformOptions {
+  varName: string; // something must be defined
+}
+
+export const renameByMappingTransformer: SynchronousDataTransformerInfo<RenameByMappingTransformOptions> = {
+  id: DataTransformerID.renameByMapping,
+  name: 'Rename by Mapping',
+  description: 'Rename fields using a text-value mapping',
+  defaultOptions: {},
+
+  operator: (options) => (source) => source.pipe(map((data) => renameByMappingTransformer.transformer(options)(data))),
+
+  transformer: (options: RenameByMappingTransformOptions) => {
+    return (data: DataFrame[]) => {
+      if (!data || !data.length) {
+        return data;
+      }
+      return renameByMapping(options, data);
+    };
+  },
+};
+
+type VarMap = {
+  [key: string]: string;
+};
+
+export function getVariablesList(): [] {
+  const templateSrv: TemplateSrv = getTemplateSrv();
+  const currentVars = templateSrv.getVariables();
+  const varsList = currentVars.map((x) => ({ label: x.label || x.name, value: x.id }));
+  return varsList;
+}
+
+function convertVariableToMapping(varName: string): VarMap {
+  const templateSrv: TemplateSrv = getTemplateSrv();
+  const thisvar = templateSrv.getVariableAtIndex(varName);
+  const mapping: VarMap = {};
+  thisvar.options.map((i) => (mapping[i.value] = i.text));
```

Comment:
accessing a variables options is not going to be allowed in the future, only it's current value (the options are an internal implementation detail) , only some variables have them.

Reply:
I see... In my latest patch which I haven't yet pushed I was checking to make sure the variable is a `VariableWithOptions`. I don't really understand why the options should be private data since they are presented to the user for selection, but if you say that is the case, it makes this path impossible.